internal/archiver/tree.go

```diff
@@ -195,26 +195,13 @@ func (t *Tree) add(fs fs.FS, target, root string, pc []string) error {
 	return nil
 }
 
-func (t Tree) String() string {
-	return formatTree(t, "")
-}
-
-// formatTree returns a text representation of the tree t.
-func formatTree(t Tree, indent string) (s string) {
-	for name, node := range t.Nodes {
-		s += fmt.Sprintf("%v/%v, root %q, path %q, meta %q\n", indent, name, node.Root, node.Path, node.FileInfoPath)
-		s += formatTree(node, indent+"    ")
-	}
-	return s
-}
-
 // unrollTree unrolls the tree so that only leaf nodes have Path set.
 func unrollTree(f fs.FS, t *Tree) error {
 	// if the current tree is a leaf node (Path is set) and has additional
 	// nodes, add the contents of Path to the nodes.
 	if t.Path != "" && len(t.Nodes) > 0 {
 		debug.Log("resolve path %v", t.Path)
-		entries, err := fs.ReadDirNames(f, t.Path)
```

Comment:
The `fs.ReadDirNames` and `readdirnames` functions have slightly different semantics. The latter sets `fs.O_NOFOLLOW` whereas the former does not. I'm right now not sure which is the correct behavior.

Reply:
I can't find any change in behavior and all the tests pass, but I can solve this by changing readdirnames's signature to

```
// flags are passed to fs.OpenFile. O_RDONLY is implied.
func readdirnames(filesystem fs.FS, dir string, flags int) ([]string, error) {
	f, err := filesystem.OpenFile(dir, flags|fs.O_RDONLY, 0)
	// ...
}
```

This is after I remove the sorted argument in favor of sorting on the spot. Is that ok?